packages/core/core/src/PackagerRunner.js

```diff
@@ -566,12 +576,17 @@ function assignComplexNameHashes(hashRefToNameHash, bundles, bundleInfoMap) {
       ...getBundlesIncludedInHash(bundle.id, bundleInfoMap),
     ];
 
-    hashRefToNameHash.set(
-      bundle.hashReference,
-      md5FromString(
-        includedBundles.map(bundleId => bundleInfoMap[bundleId].hash).join(':'),
-      ).slice(-8),
-    );
+    // we want stable names with Hot Module Reloading
+    // we want names based on content during production for cacheability
```

Comment:
Should the names be stable only with HMR or always in development mode?

Reply:
We only need it with HMR and if I change it so that it's stable in any mode but 'production' then it breaks two unit tests that expect output file names to change with content changes. If I swap those unit tests over to mode: 'production' then it breaks 40+ tests for some reason that I haven't been able to determine.

Reply:
I think ideally this would be behind a flag for content hashing, like --no-content-hash from v1. That should be false by default in development mode.

Reply:
I added no-content-hashing.

content-hashing is automatically set to false if HMR is enabled.